diagrams/request_uri_mode_post.md

```diff
@@ -0,0 +1,38 @@
+```plantuml
+@startuml
+
+autonumber
+
+participant "Wallet" as w
+
+participant "User Agent" as u
+
+participant "Verifier" as r
+
+u --> r : use
+activate r
+
+r --> u: authorization request\n(client_id, request_uri, request_uri_method=post, [client_id_scheme])
+deactivate r
+u --> w: authorization request\n(client_id, request_uri, request_uri_method=post, [client_id_scheme])
+activate w
```

Comment:
step needed here for wallet to check if client_id is trusted

Reply:
If wallet determines client_id is trusted it can go ahead and send metadata to the request uri, but if wallet determines that client_id is unknown or known and untrusted then it should ask the user 'is it OK to send metadata to the request uri"

Reply:
Add the following text
"Note that the client_id is self asserted by the verifier. If it is true and trusted then no harm will come to the user. However, if the client_id is not trusted, whether it is true or not, then the user should be informed that an untrusted verifier is requesting information and asked if he/she wants to proceed. Finally if the client_id is false but identifies a trusted verifier, then it is vital that the request_uri that is responded to is the one that actually belongs to the trusted client_id, and is provided by the trust framework, and is not the request_uri provided in the request"

Reply:
the problem here is that there is no guarantee to the client_id as it is self-asserted and the current way with custom URL schemes does not offer any way to identify the source of the request (that's why we had the initial proposal to sign this - but it seemed too complicated). If a rogue website is using a trusted client_id from a valid RP, then the wallet has no means to detect this and the wallet will release the metadata. It seems there is no way around it?

Reply:
Please read my Note again. It says that the wallet, having validated the client_id with its trust framework, must obtain the return-uri from the trust framework (and not from the incoming protocol message) and send its metadata to this endpoint. Do you see a problem with this?

Reply:
You are right @David-Chadwick. The Wallet SHOULD check that the client_id is from a Verifier in a trust framework AND that the request_uri matches to this client_id?

Reply:
@David-Chadwick I incorporated your note and extended the privacy considerations as discussed in the last WG meeting. Please review and approve.